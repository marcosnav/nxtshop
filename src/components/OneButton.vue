<template>
  <button :class="componentClass">
    <slot />
  </button>
</template>

<script lang="ts">
import Vue from 'vue';

const CLASS = {
  STD: 'OneButton',
  BIG: 'OneButton--big',
  SMALL: 'OneButton--small',
  TEXTLINE: 'OneButton--textline',
  SECONDARY: 'OneButton--secondary',
  DARK: 'OneButton--dark',
};

export default Vue.extend({
  props: {
    big: Boolean,
    small: Boolean,
    textline: Boolean,
    secondary: Boolean,
    dark: Boolean,
  },

  computed: {
    componentClass() {
      const { big, small, textline, secondary, dark } = this;
      const componentClass = [CLASS.STD];

      // Sizes
      if (big) {
        componentClass.push(CLASS.BIG);
      } else if (small) {
        componentClass.push(CLASS.SMALL);
      }

      // Types
      if (textline) {
        componentClass.push(CLASS.TEXTLINE);
      } else if (secondary) {
        componentClass.push(CLASS.SECONDARY);
      } else if (dark) {
        componentClass.push(CLASS.DARK);
      }

      return componentClass.join(' ');
    },
  },
});
</script>

<style lang="sass" scoped>
.OneButton
  background: var(--blue)
  border: none
  border-radius: 6px
  color: #fff
  font-size: 0.833em
  font-weight: 700
  line-height: 26px
  padding: 0 16px
  position: relative
  text-align: center

  &--big
    font-size: 1em
    line-height: 42px

  &--small
    border-radius: 10px
    font-size: 0.7em
    line-height: 20px
    padding: 0 10px

  &--textline
    background: transparent
    color: var(--text)
    text-decoration: underline

  &--secondary
    background: var(--subtle-gray)
    color: var(--text)

  &--dark
    background: var(--dark-blue)
</style>
