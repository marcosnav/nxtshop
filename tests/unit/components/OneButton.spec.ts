import { shallowMount } from '@vue/test-utils';
import OneButton from '@/components/OneButton.vue';

describe('<OneButton>', () => {
  test('renders properly', () => {
    const component = shallowMount(OneButton, {
      slots: {
        default: 'Add to Cart',
      },
    });
    expect(component.classes('OneButton')).toBeTruthy();
    expect(component.text()).toMatch('Add to Cart');
    expect(component.element).toMatchSnapshot();
  });

  describe('button sizes', () => {
    test('big=true', () => {
      const component = shallowMount(OneButton, {
        propsData: {
          big: true,
        },
      });
      expect(component.classes('OneButton--big')).toBeTruthy();
      expect(component.element).toMatchSnapshot();
    });

    test('small=true', () => {
      const component = shallowMount(OneButton, {
        propsData: {
          small: true,
        },
      });
      expect(component.classes('OneButton--small')).toBeTruthy();
      expect(component.element).toMatchSnapshot();
    });
  });

  describe('button types', () => {
    test('textline=true', () => {
      const component = shallowMount(OneButton, {
        propsData: {
          textline: true,
        },
      });
      expect(component.classes('OneButton--textline')).toBeTruthy();
      expect(component.element).toMatchSnapshot();
    });

    test('secondary=true', () => {
      const component = shallowMount(OneButton, {
        propsData: {
          secondary: true,
        },
      });
      expect(component.classes('OneButton--secondary')).toBeTruthy();
      expect(component.element).toMatchSnapshot();
    });
  });
});
