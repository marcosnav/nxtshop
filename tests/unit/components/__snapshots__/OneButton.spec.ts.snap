// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OneButton> button sizes big=true 1`] = `
<button
  class="OneButton OneButton--big"
/>
`;

exports[`<OneButton> button sizes small=true 1`] = `
<button
  class="OneButton OneButton--small"
/>
`;

exports[`<OneButton> button types secondary=true 1`] = `
<button
  class="OneButton OneButton--secondary"
/>
`;

exports[`<OneButton> button types textline=true 1`] = `
<button
  class="OneButton OneButton--textline"
/>
`;

exports[`<OneButton> renders properly 1`] = `
<button
  class="OneButton"
>
  Add to Cart
</button>
`;
